src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/ChoosePaymentMethodEligibilityValidatorSpec.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Command\Checkout\ChoosePaymentMethod;
+use Sylius\Bundle\ApiBundle\Command\PaymentMethodCodeAwareInterface;
+use Sylius\Bundle\ApiBundle\Validator\Constraints\ChoosePaymentMethodEligibility;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+final class ChoosePaymentMethodEligibilityValidatorSpec extends ObjectBehavior
+{
+    function let(PaymentMethodRepositoryInterface $paymentMethodRepository): void
+    {
+        $this->beConstructedWith($paymentMethodRepository);
+    }
+
+    function it_is_a_constraint_validator(): void
+    {
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_if_constraint_does_not_extend_payment_method_code_aware_interface(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', ['', new class() extends Constraint {
+            }]);
+    }
+
+    function it_throws_an_exception_if_constraint_does_not_type_of_choose_payment_method_eligibility(): void
+    {
+        $constraint = new class() extends Constraint implements PaymentMethodCodeAwareInterface {
```

Comment:
Shouldn't it be `$value` rather than `$constraint`? This way both this and the previous spec checks the same assertion in the class. I would expect to have here a proper value and bad constraint (as the previous spec clearly checks bad value)

Reply:
You can simplify this spec to sth like:
```
function it_throws_an_exception_if_constraint_is_not_an_instance_of_chosen_shipping_method_eligibility(): void
{
    $this
        ->shouldThrow(\InvalidArgumentException::class)
        ->during('validate', [new ChoosePaymentMethod('code'), new class() extends Constraint {}])
    ;
}
```